### PR TITLE
DCMAW-8264: Add Event Service

### PR DESCRIPTION
**Pull request by J-son1**
​DCMAW-8264
### What changed

Adds `EventService` class as a dependency to the credential Lambda to send messages to TxMA

### Why did it change
This is part of the work for implementing the Client Credentials Grant flow and asynchronous CRI credential requests outlined in [this TD](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3917447301/Strategic+App+DCMAW-7690+Implementing+Client+Credentials+Grant+Flow+and+Asynchronous+CRI+credential+requests)

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [x] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [ ] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: src/functions/asyncCredential/asyncCredentialHandler.ts [hunk 1]
```diff
@@ -206,19 +207,70 @@ export async function lambdaHandler(
...
     await sessionService.createSession(sessionConfig);
 
+  const eventService = dependencies.eventService(config.SQS_QUEUE);
+
   if (sessionServiceCreateSessionResult.isError) {
+    const writeEventResult = await eventService.writeEvent({
+      eventName: "DCMAW_ASYNC_CRI_5XXERROR" as EventName,
+      sub,
+      sessionId,
+      ipAddress: "",
```
Comment (J-son1) [inline src/functions/asyncCredential/asyncCredentialHandler.ts:217]: Should this be removed if this is not an user interaction event or is there still value in having this field?

File: src/functions/services/events/sqsClient.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { SQSClient } from "@aws-sdk/client-sqs";
+import { NodeHttpHandler } from "@smithy/node-http-handler";
+
+export const sqsClient = new SQSClient({
+  region: process.env.REGION,
+  maxAttempts: 2,
```
Comment (jamesamooney) [inline src/functions/services/events/sqsClient.ts:6]: 3?

File: src/functions/asyncCredential/asyncCredentialHandler.ts [hunk 3]
```diff
@@ -206,20 +211,71 @@ export async function lambdaHandler(
...
+    logger.log("ERROR_WRITING_AUDIT_EVENT", {
+      errorMessage: "Unexpected error writing the DCMAW_ASYNC_CRI_START event",
+    });
+  }
+
+  logger.log("SESSION_CREATED");
   return sessionCreatedResponse(parsedRequestBody.sub);
 }
 
+interface IBaseEventConfig {
```
Comment (J-son1) [inline src/functions/asyncCredential/asyncCredentialHandler.ts:260]: Should this move to sqsClient.ts?

File: src/functions/services/events/eventService.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+export interface IEventConfig {
+  eventName: EventName;
+  sub: string;
+  sessionId: string;
+  ipAddress: string;
+  govukSigninJourneyId: string;
+  clientId: string;
+  getNowInMilliseconds: () => number;
+  componentId: string;
```
Comment (J-son1) [inline src/functions/services/events/eventService.ts:82]: If the `baseEventCofig` is moved to this file, could it be used here to replace these fields?

File: src/functions/asyncCredential/asyncCredentialHandler.ts (hunk 3 above)
Comment (jamesamooney) [reply src/functions/asyncCredential/asyncCredentialHandler.ts]: Updated

File: src/functions/services/events/eventService.ts (hunk 4 above)
Comment (jamesamooney) [reply src/functions/services/events/eventService.ts:85]: Updated

File: src/functions/services/events/sqsClient.ts (hunk 2 above)
Comment (jamesamooney) [reply src/functions/services/events/sqsClient.ts]: Done

File: src/functions/asyncCredential/asyncCredentialHandler.ts (hunk 1 above)
Comment (jamesamooney) [reply src/functions/asyncCredential/asyncCredentialHandler.ts]: Removed

**Pull request merged**